cmd/installer/pkg/bootloader/syslinux/adv.go

```diff
@@ -0,0 +1,195 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package syslinux
+
+import (
+	"encoding/binary"
+	"fmt"
+	"io"
+)
+
+const (
+	// AdvSize is the total size.
+	AdvSize = 512
+	// AdvLen is the usable data size.
+	AdvLen = AdvSize - 3*4
+	// AdvMagic1 is the head signature.
+	AdvMagic1 = uint32(0x5a2d2fa5)
+	// AdvMagic2 is the total checksum.
+	AdvMagic2 = uint32(0xa3041767)
+	// AdvMagic3 is the tail signature.
+	AdvMagic3 = uint32(0xdd28bf64)
+)
+
+const (
+	// AdvEnd is the noop tag.
+	AdvEnd = iota
+	// AdvBootonce is the bootonce tag.
+	AdvBootonce
+	// AdvMenusave is the menusave tag.
+	AdvMenusave
+	// AdvUpgrade is the upgrade tag.
+	AdvUpgrade
```

Comment:
We should add a buffer of maybe 2 or 3 "reserved" slots here, just in case syslinux decides to add moer flags in the furture.